optimum/utils/preprocessing/image_classification.py

```diff
@@ -1,114 +1,115 @@
-
-    def run_evaluation(self, eval_dataset: Dataset, pipeline: ImageClassificationPipeline, metrics: List[str]):
-        all_metrics = combine(metrics)
-
-        task_evaluator = evaluator("image-classification")
-
-        results = task_evaluator.compute(
-            model_or_pipeline=pipeline,
-            data=eval_dataset,
-            metric=all_metrics,
-            input_column=self.data_keys["primary"],
-            label_column=self.ref_keys[0],
-            label_mapping=self.config.label2id,
+    def dataset_processing_func(
+        self, example: Dict[str, Any], data_keys: Dict[str, str], ref_keys: Optional[List[str]] = None
+    ) -> Dict[str, Any]:
+        image = example[data_keys["image"]]
+        example["pixel_values"] = self.transforms(image.convert("RGB")).to(torch.float32).numpy()
+        return example
+
+    def try_to_guess_data_keys(self, column_names: List[str]) -> Optional[Dict[str, str]]:
+        image_key_name = None
+        for name in column_names:
+            if "image" in name:
+                image_key_name = name
+                break
+        if image_key_name is None and len(column_names) == 2:
+            if "label" in column_names[0]:
+                image_key_name = column_names[1]
+            elif "label" in column_names[1]:
+                image_key_name = column_names[0]
+
+        return {"image": image_key_name} if image_key_name is not None else None
+
+    def try_to_guess_ref_keys(self, column_names: List[str]) -> Optional[List[str]]:
+        for name in column_names:
+            if "label" in name:
+                return [name]
+
+    def load_dataset(
+        self,
+        path: str,
+        data_keys: Optional[Dict[str, str]] = None,
+        ref_keys: Optional[List[str]] = None,
+        only_keep_necessary_columns: bool = False,
+        load_smallest_split: bool = False,
+        num_samples: Optional[int] = None,
+        shuffle: bool = False,
+        **load_dataset_kwargs,
+    ) -> Union["DatasetDict", "Dataset"]:
+        dataset = super().load_dataset(
+            path,
+            data_keys=data_keys,
+            ref_keys=ref_keys,
+            only_keep_necessary_columns=only_keep_necessary_columns,
+            load_smallest_split=load_smallest_split,
+            num_samples=num_samples,
+            shuffle=shuffle,
+            **load_dataset_kwargs,
         )
-
-        return results
-
-    def get_metrics(self, predictions: List, references: List, metric: Metric):
-        metrics_res = metric.compute(predictions=predictions, references=references)
-
-        # `metric.compute` may return a dict or a number
-        if not isinstance(metrics_res, dict):
-            metrics_res = {metric.name: metrics_res}
-
-        return metrics_res
-
-    def get_pipeline_kwargs(self):
-        return {}
+        # TODO: do we want to do that here?
+        # eval_dataset = eval_dataset.align_labels_with_mapping(self.config.label2id, self.ref_keys[0])
```

Comment:
As long as you don't evaluate it's fine. If you do evaluation, this is needed as some models on the Hub don't have there label2id matching the datasets label2id (different ordering).